### PR TITLE
love -> 0.10 added

### PR DESCRIPTION
**Pull request by leenaars**
LÖVE has different versions with partially incompatible API's. This commit adds the latest stable upstream version. I don't know what the canonical way of dealing with this is, so I left the oldest version (0.8) intact as "default". 0.9 has been added before by someone else in a similar way..

### CODE REVIEW

**Comment by 7c6f434c** [comment]
So you add 0.8.nix but never use it?

**Comment by jgillich** [comment]
Please use `fetchFromBitbucket`.

File: pkgs/development/interpreters/love/0.10.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  configureFlags = [
+    "--with-lua=luajit"
+  ];
+
+  meta = {
+    homepage = "http://love2d.org";
+    description = "A Lua-based 2D game engine/scripting language";
+    license = stdenv.lib.licenses.zlib;
+
```
Comment (jagajaga) [inline pkgs/development/interpreters/love/0.10.nix]: Please remove this empty line.

**Comment by jagajaga** [comment]
We do already have https://github.com/NixOS/nixpkgs/tree/master/pkgs/development/interpreters/love.

**Comment by jagajaga** [comment]
So you can just update the default version to `0.10`. I don't know why do we need all of the versions.

**Comment by leenaars** [comment]
Meanwhile (5 hours ago) a new version 0.10.1 was released. This was added to default.nix instead of 0.10.0. The games created with this lib are kind of specific for that version, and some interesting games don't follow the API upgrades so having multiple versions of LOVE around makes sense. Can't have too much LOVE anyway ;)

Used fetchFromBitbucket instead of direct fetchurl.

**Comment by jagajaga** [comment]
Ok, let's wait for travis's result :)

**Comment by leenaars** [comment]
Not only was there a new point release, but this also involved a new automagic based toolchain to preconfigure. Added this. The different versions allow people building packages with these different versions to pin to a specific version, as their API's are different and not every game will be updated.

**Comment by jagajaga** [comment]
You don't need to define `0.10` as a separate file if it is in `default.nix`. So please remove `default.nix` and in `all-packages` do
```
love_0_8 = callPackage ../development/interpreters/love/0.8.nix { };
love_0_10 = callPackage ../development/interpreters/love/0.10.nix { };
love = love_0_10;
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -5221,9 +5221,11 @@ let
 
   lolcode = callPackage ../development/interpreters/lolcode { };
 
-  love = callPackage ../development/interpreters/love {lua=lua5_1;};
   love_luajit = callPackage ../development/interpreters/love {lua=luajit;};
+  love_0_8 = callPackage ../development/interpreters/love/0.8.nix { };
   love_0_9 = callPackage ../development/interpreters/love/0.9.nix { };
+  love_0_10 = callPackage ../development/interpreters/love/0.10.nix { };
+  love = love_0_10
```
Comment (jagajaga) [inline pkgs/top-level/all-packages.nix]: Missing `;`.

**Comment by jagajaga** [comment]
Don't forget to delete `default.nix`.

**Comment by leenaars** [comment]
@jagajaga: removed it, I thought is was "mandatory" to always have a default.nix in a folder.
I by the way also removed love_luajit entry, after grepping through nixpkgs and not finding a single mention. Especially since all current versions *are* using luajit already...

**Pull request merged**